packages/babel-helper-annotate-as-pure/src/index.js

```diff
@@ -1,12 +1,15 @@
 import * as t from "@babel/types";
+import type { Node } from "@babel/types";
 
 const PURE_ANNOTATION = "#__PURE__";
 
-const isPureAnnotated = ({ leadingComments }) =>
+const isPureAnnotated = ({ leadingComments }: Node): boolean =>
   !!leadingComments &&
   leadingComments.some(comment => /[@#]__PURE__/.test(comment.value));
 
-export default function annotateAsPure(pathOrNode) {
+export default function annotateAsPure(
+  pathOrNode: Node | { node: Node },
```

Comment:
Since this function does not use `Path.prototype`, I type the input as `{ node: Node }`.